geoh5py/shared/entity.py

```diff
@@ -50,27 +67,41 @@ class Entity(ABC):
         "Public": "public",
         "Visible": "visible",
     }
-    _visible = True
-    _default_name: str = "Entity"
-
-    def __init__(self, uid: uuid.UUID | None = None, **kwargs):
-        self._uid = (
-            str2uuid(uid) if isinstance(str2uuid(uid), uuid.UUID) else uuid.uuid4()
-        )
-
-        self._allow_delete = True
-        self._allow_move = True
-        self._allow_rename = True
-        self._clipping_ids: list[uuid.UUID] | None = None
-        self._metadata: dict | None = None
-        self._name = self._default_name
-        self._on_file = False
-        self._parent: Entity | None = None
-        self._partially_hidden = False
-        self._public = True
-
+    _default_name: str | None = None
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        entity_type: shared.EntityType | None = None,
+        allow_delete: bool = True,
+        allow_move: bool = True,
+        allow_rename: bool = True,
+        clipping_ids: list[uuid.UUID] | None = None,
+        metadata: dict | None = None,
+        name: str | None = None,
+        on_file: bool = False,
+        partially_hidden: bool = False,
+        parent: EntityContainer | None = None,
+        public: bool = True,
+        uid: uuid.UUID | None = None,
+        visible: bool = True,
+        **kwargs,
+    ):
+        self.on_file = on_file
+        self.name = name or self._default_name or type(self).__name__
+        self._entity_type = self.validate_entity_type(entity_type)
+        self.uid: uuid.UUID = uid or uuid.uuid4()
+        self.allow_delete = allow_delete
+        self.allow_move = allow_move
+        self.allow_rename = allow_rename
+        self.clipping_ids = clipping_ids
+        self.metadata = metadata
+        self.parent = parent or self.workspace.root
+        self.partially_hidden = partially_hidden
+        self.public = public
+        self.visible = visible
+
+        # TODO Deprecate in favor of explicit attribute setter
```

Comment:
do we still need the map_attributes then?

Reply:
Will be removed once we tackle GEOPY-1668